src/app/navigation/utils/index.ts

```diff
@@ -61,8 +62,9 @@ export const findRouteMatch = (routes: AnyRoute[], location: Location | string):
       const keys: Key[] = [];
       const re = pathToRegexp(path, keys, {end: true});
       const match = re.exec(typeof location === 'string' ? location : location.pathname);
+      const search = typeof location.search === 'string' ? location.search : '';
       if (match) {
-        return formatRouteMatch(route, (typeof location !== 'string' && location.state) ?? {}, keys, match);
+        return formatRouteMatch(route, (typeof location !== 'string' && location.state) ?? {}, keys, match, search);
```

Comment:
@TomWoodward without the changes to this file, the `?message=retired` redirects will never hit `matchSearch` because there is no search param on the match

Reply:
(this is re your suggestion to incorporate your changes to matchSearch and remove my change to formatRouteMatch)

Reply:
well in that case the target is still a rex page, i guess the issue is you don't want to flash the previous book before switching to the new book....

can we like.... ignore that a little bit for now? maybe a better solution would be to rearrange the resolveContent hook to not dispatch receiveBook if there is a redirect

Reply:
I can remove the `services.history.replace(to)` from processBrowserRedirect if that's what you mean - I don't think the flash is a big deal. could rearrange resolveContent but that will take more time.

Reply:
i think just remove the replace and call it good for now, unless you've already done more

Reply:
it's removed